### PR TITLE
Add examples for hack lang (thanks @azjezz)

### PR DESCRIPTION
**Pull request by azjezz**

### CODE REVIEW

**Comment by azjezz** [comment]
cc @fredemmott @usox

**Comment by asergeyev** [comment]
But entrypoint thing stays then? Will it work @usox ?

I second desire to ask for rebase/squash and have less commits, could you do this @azjezz?    I can squash it on merge but that'll stripe authoring.

**Comment by azjezz** [comment]
@usox @asergeyev `__EntryPoint` attribute works as of 3.30 which is the version stated in the README.
i would say squash and merge :P

**Comment by asergeyev** [comment]
Thanks

File: examples/hack/count.hh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+<<__EntryPoint>>
+async function count_to_ten(): Awaitable<noreturn> {
+  $output = request_output();
+  for ($count = 1; $count <= 10; $count++) {
+    await $output->writeAsync(
+      Str\format("%d\n",$count)
+    );
+ 
+    // usleep is builtin, it is not an async builtin - so it also must block the main request thread
+    usleep(500000);
```
Comment (fredemmott) [inline examples/hack/count.hh]: HH\Asio\usleep()

Comment (fredemmott) [reply examples/hack/count.hh]: as a side note, async is not multithreading. Some async operations may use threading as an implementation detail, but in general (and for IO specifically), it's single-threaded.

Comment (azjezz) [reply examples/hack/count.hh]: fixed here : #340

**Pull request merged**